static/js/src/services/code.js

```diff
@@ -1,29 +1,52 @@
 'use strict';
 
-function code(commands, blockly) {
-  return { generate, execute };
+function code(commands, blockly, Boards) {
+  return { generate, execute, setWorkshop };
+  var currentWorkshop = null;
+
+  // TODO - this seems pretty wrong. Let's ask Brian or someone who knows Angular
+  // pretty well to take a look!
+  function setWorkshop(workshop) {
+    currentWorkshop = workshop;
+  }
```

Comment:
Hey @BrianGenisio - would you mind taking a look at this and letting me know if there's a better way of doing it Angular? Basically, this is the code generator service, but it needs some context about the current workshop that is loaded. I know you can't really inject the scope into a service. I know giving a service a state like this is pretty bad too. I'd rather not have to pass in the workshop object into each generate or compile call. Is there some nice way of doing this you can think of?